src/Compilers/CSharp/Portable/Symbols/Source/QuickAttributeChecker.cs

```diff
@@ -145,5 +145,53 @@ internal enum QuickAttributes : byte
         AssemblyKeyName = 1 << 2,
         AssemblyKeyFile = 1 << 3,
         AssemblySignatureKey = 1 << 4,
+        Last = AssemblySignatureKey,
+    }
+
+    internal static class QuickAttributeHelpers
+    {
+        /// <summary>
+        /// Returns the <see cref="QuickAttributes"/> that corresponds to the particular type 
+        /// <paramref name="name"/> passed in.  If <paramref name="inAttribute"/> is <see langword="true"/>
+        /// then the name will be checked both as-is as well as with the 'Attribute' suffix.
+        /// </summary>
+        public static QuickAttributes GetQuickAttributes(string name, bool inAttribute)
+        {
+            // Update this code if we add new quick attributes.
+            Debug.Assert(QuickAttributes.Last == QuickAttributes.AssemblySignatureKey);
+
+            var result = QuickAttributes.None;
+            if (Matches(AttributeDescription.TypeIdentifierAttribute))
+                result |= QuickAttributes.TypeIdentifier;
+            else if (Matches(AttributeDescription.TypeForwardedToAttribute))
+                result |= QuickAttributes.TypeForwardedTo;
+            else if (Matches(AttributeDescription.AssemblyKeyNameAttribute))
+                result |= QuickAttributes.AssemblyKeyName;
+            else if (Matches(AttributeDescription.AssemblyKeyFileAttribute))
+                result |= QuickAttributes.AssemblyKeyFile;
+            else if (Matches(AttributeDescription.AssemblySignatureKeyAttribute))
+                result |= QuickAttributes.AssemblySignatureKey;
+
+            return result;
+
+            bool Matches(AttributeDescription attributeDescription)
```

Comment:
>Matches [](http://example.com/codeflow?start=17&length=7)

nit: compiler uses lower-case for local functions

Reply:
I'm surprised the correctness leg didn't catch that. I thought we'd set that up in editorconfig

Reply:
i don't believe hte compiler overrides the codestyle rules we have here, so the default is to not check this at all.

Reply:
but i will rename this.